### DISCUSSION TITLE
Weeks in the date format

### DISCUSSION THREAD

**Discussion by vanillajonathan** [PRQL/prql#3548] [Ideas]
The [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) standard have some formats which PRQL does not understand.

- `20230920T113041Z` (without hyphens and colons).
- `2023‐W38` (week 38)
- `2023‐W38‐3` (week with weekday)
- `2023‐263` (ordinal date)

**Comment by max-sixty** [comment]
Yes agree!

Do DBs support these?

At the moment we mostly pass strings through. One mini-project would be to parse the dates ourselves (likely using a rust library) into an internal representation, and then write them out as a string. This would allow for great error messages. I know a common complaint for frequent-but-not-expert SQL users is that it's difficult to understand how to write date literals correctly.

**Reply by vanillajonathan** [reply]
I don't know.

**Reply by vanillajonathan** [reply]
I also noticed that PRQL handles timezones in the `-05:00` and `-0500` format, but not the `-05` format.

**Reply by max-sixty** [reply]
Very happy to add that, assuming it's consistent with the ISO (I think it is). And the parsing shouldn't be too difficult given it's making things optional...

**Reply by vanillajonathan** [reply]
I think it is. https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators

**Reply by max-sixty** [reply]
Great, we could definitely add this, then
